### PR TITLE
Add 8ball app and change boxclk to use clockbackgrounds app

### PR DESCRIPTION
**Pull request by only-meeps**
added an app called 8ball (a magic 8ball) and updated the clock boxclk to use the new app clockbackgrounds

### CODE REVIEW

**Comment by only-meeps** [comment]
fixed the warnings

**Review by bobrippling — COMMENTED** [review]
Had a few questions :)

File: apps/boxclk/app.js [hunk 1]
```diff
@@ -1,415 +1,402 @@
...
-
-  let boxKeys = Object.keys(boxes);
-
-  boxKeys.forEach((key) => {
-    let boxConfig = boxes[key];
-    boxPos[key] = {
-      x: w * boxConfig.boxPos.x,
-      y: h * boxConfig.boxPos.y
+let background = require("clockbg");
+Modules.addCached("widget_utils", function() {
```
[the remark this answers is not in the document]
Comment (only-meeps) [reply apps/boxclk/app.js:2]: yes, I did copy it out of the web ide. I think I took the code off the watch first though

File: apps/boxclk/metadata.json [hunk 2]
```diff
@@ -4,6 +4,7 @@
   "version": "0.05",
```
[the remark this answers is not in the document]
Comment (only-meeps) [reply apps/boxclk/metadata.json:4]: ah, it looks like I did change the version but the build failed. I will try it again soon

File: apps/boxclk/app.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (only-meeps) [reply apps/boxclk/app.js:2]: Unfortunately, the build failed
Error: App boxclk's boxclk.app.js is a JS file but isn't valid JS

File: apps/8ball/app.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
+try {keyboard = require(keyboard);} catch(e) {keyboard = null;}var keyboard = "textinput";
```
[the remark this answers is not in the document]
Comment (only-meeps) [reply apps/8ball/app.js]: fixed!

File: apps/boxclk/metadata.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (only-meeps) [reply apps/boxclk/metadata.json:4]: Fixed!

[the remark this answers is not in the document]
Comment (bobrippling) [reply apps/boxclk/metadata.json:4]: nice!

File: apps/boxclk/app.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (bobrippling) [reply apps/boxclk/app.js:2]: ah yes - so when building an app the app loader will add a few things in (the `Modules.addCached(...)` bit, for example). I can drop them out so it includes just your changes, unless you can see what I mean and want to have a go?

[the remark this answers is not in the document]
Comment (only-meeps) [reply apps/boxclk/app.js:2]: can you do it? I'm kinda new here and I don't know how to

[the remark this answers is not in the document]
Comment (bobrippling) [reply apps/boxclk/app.js]: Sure, done - could you check the changes you want are still in boxclk? If you're happy I'll then merge

[the remark this answers is not in the document]
Comment (only-meeps) [reply apps/boxclk/app.js]: yes, still works

[the remark this answers is not in the document]
Comment (only-meeps) [reply apps/boxclk/app.js]: thank you

[the remark this answers is not in the document]
Comment (only-meeps) [reply apps/boxclk/app.js]: also can I ask : does your watch sometimes just refuse to connect to your computer with bluetooth? is there any way to fix this?

[the remark this answers is not in the document]
Comment (bobrippling) [reply apps/boxclk/app.js]: I do get that occasionally - my watch's bluetooth ends up turned off and I have to either toggle it or do the "Make Connectable" option in settings, but perhaps there's something odd going on there. @gfwilliams does this ring any bells?

**Pull request merged**